### PR TITLE
Build and test the examples

### PR DESCRIPTION
**Pull request by nipunayf**
# Description

The PR improves the daily build, enabling it to support the building and testing of the example programs found in the root directory. Additionally, the PR made the following improvements:

* Added Checkstyle, SpotBugs, and JaCoCo workflows to the Gradle build.
* The directory structure will be adhered to according to the document: https://github.com/ballerina-platform/ballerina-extended-library/blob/main/docs/module-contribution-guide.md
* Ignored vulnerable transitive Java dependencies to resolve the Trivy scan failure.

Related to https://github.com/ballerina-platform/ballerina-extended-library/issues/539

### CODE REVIEW

File: native/src/main/java/org/ballerinax/asb/util/ASBUtils.java [hunk 1]
```diff
@@ -364,24 +358,27 @@ private static Object getValueWithIntendedTypeRecursive(byte[] value, Type type)
         return intendedValue;
     }
 
-    private static boolean hasStringType(UnionType type) {
-        return type.getMemberTypes().stream().anyMatch(memberType -> memberType.getTag() == STRING_TAG);
-    }
-
```
Comment (nipunayf) [inline native/src/main/java/org/ballerinax/asb/util/ASBUtils.java:370]: @NipunaRanasinghe The SpotBugs validation detects this as an unused private function. Is there a reason to have it? If so, I can exclude it by adding a filter.

File: examples/build.gradle [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+apply plugin: 'java'
+
+description = 'Ballerina - Azure-service-bus Examples'
```
Comment (NipunaRanasinghe) [inline examples/build.gradle:21]:
```suggestion
description = 'Ballerina - Azure Service Bus Examples'
```

File: native/build.gradle [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+ */
+
+plugins {
+    id 'java'
+    id 'com.github.spotbugs'
+    id 'checkstyle'
+    id 'jacoco'
+}
+
+description = 'Ballerina - Azure-service-bus Native'
```
Comment (NipunaRanasinghe) [inline native/build.gradle:26]:
```suggestion
description = 'Ballerina - Azure Service Bus Native'
```

File: native/src/main/java/org/ballerinax/asb/util/ASBUtils.java (hunk 1 above)
Comment (NipunaRanasinghe) [reply native/src/main/java/org/ballerinax/asb/util/ASBUtils.java:370]: +1 to remove

File: .github/workflows/dev-stg-release.yml [hunk 4]
```diff
@@ -7,71 +7,63 @@ on:
...
-          
+
+      # Push to Ballerina Staging Central
       - name: Push to Staging
         if: github.event.inputs.bal_central_environment == 'STAGE'
-        uses: ballerina-platform/ballerina-action/@2201.7.0
-        with:
-          args:
-            push
+        run: bal push
```
Comment (NipunaRanasinghe) [inline .github/workflows/dev-stg-release.yml:56]: It seems we dont have a `setup-ballerina` action in this workflow

**Review by NipunaRanasinghe — APPROVED** [review]

**Pull request merged**